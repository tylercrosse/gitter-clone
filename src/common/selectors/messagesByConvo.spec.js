import makeGetMessagesByConvoId, {
  burstify } from './messagesByConvo';

const setup = (overrides) => (
  Object.assign({
    messages: {
      0: {
        id: 0,
        createdAt: '2017-02-01T09:00:00-08:00',
        text: 'Run the tests',
        convo: 'chat',
      },
      1: {
        id: 1,
        createdAt: '2017-02-01T09:01:00-08:00',
        text: 'Use Redux',
        convo: 'chat',
      },
      2: {
        id: 2,
        createdAt: '2017-02-01T09:10:00-08:00',
        text: 'Fix the tests',
        convo: 'react',
      },
      3: {
        id: 2,
        createdAt: '2017-02-01T09:20:00-08:00',
        text: 'Fix the tests',
        convo: 'chat',
      }
    }
  }, overrides));

describe('getMessagesByConvoId selector', () => {
  it('should select desired messages from state', () => {
    const getMessagesByConvoId = makeGetMessagesByConvoId();
    expect(getMessagesByConvoId(setup(), 'chat'))
      .toMatchSnapshot();
    expect(getMessagesByConvoId(setup(), 'react'))
      .toMatchSnapshot();
  });

  it('should burstify messages', () => {
    const { messages } = setup();
    const msgA = Object.values(messages);
    expect(burstify(msgA)).toMatchSnapshot();
  });
});
