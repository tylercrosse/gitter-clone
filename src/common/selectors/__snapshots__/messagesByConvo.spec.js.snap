// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMessagesByConvoId selector should burstify messages 1`] = `
Array [
  Object {
    "burstStart": true,
    "convo": "chat",
    "createdAt": "2017-02-01T09:00:00-08:00",
    "id": 0,
    "text": "Run the tests",
  },
  Object {
    "burstStart": false,
    "convo": "chat",
    "createdAt": "2017-02-01T09:01:00-08:00",
    "id": 1,
    "text": "Use Redux",
  },
  Object {
    "burstStart": false,
    "convo": "react",
    "createdAt": "2017-02-01T09:10:00-08:00",
    "id": 2,
    "text": "Fix the tests",
  },
  Object {
    "burstStart": true,
    "convo": "chat",
    "createdAt": "2017-02-01T09:20:00-08:00",
    "id": 2,
    "text": "Fix the tests",
  },
]
`;

exports[`getMessagesByConvoId selector should select desired messages from state 1`] = `
Array [
  Object {
    "id": 0,
    "burstStart": true,
    "convo": "chat",
    "createdAt": "2017-02-01T09:00:00-08:00",
    "text": "Run the tests",
  },
  Object {
    "id": 1,
    "burstStart": false,
    "convo": "chat",
    "createdAt": "2017-02-01T09:01:00-08:00",
    "text": "Use Redux",
  },
  Object {
    "id": 2,
    "burstStart": true,
    "convo": "chat",
    "createdAt": "2017-02-01T09:20:00-08:00",
    "text": "Fix the tests",
  },
]
`;

exports[`getMessagesByConvoId selector should select desired messages from state 2`] = `
Array [
  Object {
    "id": 2,
    "burstStart": true,
    "convo": "react",
    "createdAt": "2017-02-01T09:10:00-08:00",
    "text": "Fix the tests",
  },
]
`;
